arkouda/numeric.py

```diff
@@ -3,13 +3,56 @@
 from typing import Tuple, Union
 from arkouda.client import generic_msg
 from arkouda.dtypes import *
+from arkouda.dtypes import _as_dtype
 from arkouda.pdarrayclass import pdarray, create_pdarray
 from arkouda.pdarraysetops import unique
 from arkouda.strings import Strings
 
-__all__ = ["abs", "log", "exp", "cumsum", "cumprod", "sin", "cos", 
+__all__ = ["cast", "abs", "log", "exp", "cumsum", "cumprod", "sin", "cos", 
            "where", "histogram", "value_counts"]    
 
+@typechecked
+def cast(pda : Union[pdarray, Strings], dt) -> Union[pdarray, Strings]:
+    """
+    Cast an array to another dtype.
+
+    Parameters
+    ----------
+    pda : pdarray or Strings
+        The array of values to cast
+    dtype : np.dtype or str
+        The target dtype to cast values to
+
+    Returns
+    -------
+    pdarray or Strings
+        Array of values cast to desired dtype
+
+    Notes
+    -----
+    The cast is performed according to Chapel's casting rules and is NOT safe 
+    from overflows or underflows. The user must ensure that the target dtype 
+    has the precision and capacity to hold the desired result.
+    """
+
+    if isinstance(pda, pdarray):
+        name = pda.name
+        objtype = "pdarray"
+    elif isinstance(pda, Strings):
+        name = '+'.join((pda.offsets.name, pda.bytes.name))
+        objtype = "str"
+    else:
```

Comment:
This clause is not needed as @typechecked confirms the pda is a pdarray or Strings

Reply:
Good reminder! Still getting used to it :) Fixed just now.